functions/bouncer/src/middleware/stripeEventCsCompleted.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-console */
+import debug from 'debug';
+import * as Sentry from '@sentry/node';
+import { csCompletedValidate, csMetadataValidate } from '../validate/stripe';
+import getExpandedCheckoutSession from '../lib/stripe/getExpandedCheckoutSession';
+
+const dlog = debug('that:api:functions:bouncer:stripeEventCsCompletedMw');
+
+export default async function stripeEventCsCompleted(req, res, next) {
+  dlog('stripeEventCsCompleted middleware called');
+
+  const { whRes, stripeEvent } = req;
+  whRes.stages.push('stripeEventCsCompleted');
+
+  if (stripeEvent.type !== 'checkout.session.completed') {
+    dlog('next(), not type checkout.session.completed ');
+    return next();
+  }
+
+  let expCheckoutSession;
+  try {
+    expCheckoutSession = await getExpandedCheckoutSession(
+      stripeEvent.data.object.id,
+    );
+  } catch (err) {
+    return next(err);
+  }
+
+  // replace checkout session in event object with expanded version.
+  stripeEvent.data.object = expCheckoutSession;
+
+  return Promise.all([
+    csCompletedValidate(stripeEvent).catch(err => {
+      Sentry.captureException(err);
+      return {
+        name: 'csCompletedValidate',
+        status: 400,
+        err,
+      };
+    }),
+    csMetadataValidate(stripeEvent).catch(err => {
+      Sentry.captureException(err);
+      return {
+        name: 'csMetadataValidate',
+        status: 200,
+        err,
+      };
+    }),
+  ])
+    .then(result => {
+      const errors = result.reduce((acc, cur) => {
+        if (cur.name) {
+          acc.push(cur);
+        }
+        return acc;
+      }, []);
+      if (errors.length > 0) {
+        const [value] = errors;
+        console.log(`error validating ${value.name}`);
```

Comment:
dlog?

Reply:
This is gcloud function and we explicitly want this message (and others like it) to end up in the functions logs for troubleshooting assistance.